### ISSUE TITLE
False positive: USE_AFTER_LIFETIME lang=C++ simple reproducer with exception.

### ISSUE THREAD

**Issue by thatsafunnyname** [facebook/infer#953]
Hello and thank you for infer,
  I ran `infer` on a code base and think it is reporting a false positive, I have narrowed down the code into a simple reproducer which I include below.  Thanks.

OS and versions:

```
> uname -a
Linux 4.4.0-87-generic #110-Ubuntu SMP x86_64 x86_64 x86_64 GNU/Linux

> infer --version
Infer version v0.15.0
Copyright 2009 - present Facebook. All Rights Reserved.

> clang++ --version
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
```

infer run:

```
> make clean ; infer run -- make
...
Found 1 issue
nova.cc:23: error: USE_AFTER_LIFETIME
  Variable number_five is used at line line 23, column 12 after its lifetime ended at line 21, column 19.
  21.       return squash();
  22.     } catch (std::logic_error&) {
  23. >     return 0;
  24.     }
  25.     return 0;
Summary of the reports
  USE_AFTER_LIFETIME: 1
```

Source:

```
> cat nova.cc
#include <stdexcept>
#include <iostream>

bool squash() {
  throw (std::logic_error("dead"));
}

struct Robot {
  Robot(int o) : o(o) {
    std::cerr << "Robot number " << o << ": Is alive!" << std::endl;
  }
  ~Robot() {
    std::cerr << "Robot number " << o << ": No disassemble!"  << std::endl;
  }
  int o;
};

int main() {
  Robot number_five(5);
  try {
    return squash();
  } catch (std::logic_error&) {
    return 0;
  }
  return 0;
}
```

Makefile:

```
> cat Makefile
nova:nova.cc
        clang++ -o nova nova.cc

clean:
        rm -f nova
```
 
Output:

```
> ./nova
Robot number 5: Is alive!
Robot number 5: No disassemble!
```

**Comment by jvillard** [comment]
Thanks for the report @thatsafunnyname! This seems fixed in latest master (by 84cfd0a450e0be6fe8107da9c9153e603b25f5dd I believe).

**Comment by thatsafunnyname** [comment]
Great.  Thank you.
